mono/io-layer/sockets.c

```diff
@@ -1201,7 +1201,7 @@ WSAIoctl (guint32 fd, gint32 command,
 }
 
 #ifndef PLATFORM_PORT_PROVIDES_IOCTLSOCKET
-int ioctlsocket(guint32 fd, gint32 command, gpointer arg)
+int ioctlsocket(guint32 fd, unsigned long command, gpointer arg)
```

Comment:
The values we pass in there are arguments to `ioctl()`, which does take an `unsigned long`.  `gint32` is in any case not big enough - one of the values we pass actually overflows on Darwin.  Should we use `guint64`?